src/middlewares/error.middleware.ts

```diff
@@ -1,32 +1,78 @@
+import { DEV_ENV } from '@config';
+import { ErrorsMessages } from '@constants/errorMessages';
+import { HttpStatusCode } from '@constants/httpStatusCode';
+import { ValidationError } from 'class-validator';
 import { Request, Response } from 'express';
 import {
-  Middleware,
-  ExpressErrorMiddlewareInterface
+  ExpressErrorMiddlewareInterface,
+  Middleware
 } from 'routing-controllers';
-import { DEV_ENV } from '@config';
 import { Service } from 'typedi';
 
 interface ErrorInterface extends Error {
+  name: string;
   httpCode: number;
+  description: string | undefined;
+  errors: string[] | undefined;
 }
 
 @Middleware({ type: 'after' })
 @Service()
 export class ErrorMiddleware implements ExpressErrorMiddlewareInterface {
-  error(
-    error: ErrorInterface,
+  public error(
+    error: any,
     _req: Request,
     res: Response,
-    _next: (err?: ErrorInterface) => ErrorInterface
+    _next: (err?: any) => any
   ) {
-    if (DEV_ENV) {
-      console.error(`Message: ${error.message}`);
-      console.error(`Stack: ${error.stack}`);
+    const responseObject = {} as ErrorInterface;
+
+    // If the error comes from the class-validator then it's an array of ValidationError
+    if (
+      Array.isArray(error.errors) &&
+      error.errors.every(element => element instanceof ValidationError)
+    ) {
+      res.status(HttpStatusCode.BAD_REQUEST);
+      // I think we shoudn't send the code as a field. We have HTTP for that.
+      // Also this generate code duplication.
+      responseObject.httpCode = HttpStatusCode.BAD_REQUEST;
+      responseObject.name = ErrorsMessages.BAD_REQUEST_ERROR;
+      responseObject.description = ErrorsMessages.BODY_ERRORS;
+      responseObject.errors = [];
+
+      error.errors.forEach((element: ValidationError) => {
```

Comment:
this transformation could be a little expensive with large `dto's`, isn't the default `ValidationError` coming from `class-validator` a good first step message?

Reply:
This is a good point, but the message that class-validator gives you by default its the following:

`Invalid body, check 'errors' property for more info.`

I think that we are trying to accomplish a more descriptive message to the API's consumer, therefore I would prefer to have an specific information about which fields are wrong and why like so

```javascript
{
  "httpCode": 400,
  "name": "Bad request error",
  "description": "You have errors in your request's body.Check 'errors' field for more details.",
  "errors": [
    "Property Password is too short, the minimum length is 6 characters."
  ]
}```